client/src/views/Dashboard/Dashboard.js

```diff
@@ -35,35 +40,87 @@ const Dashboard = props => {
     'Due Date',
     'Status'
   ]
+
+  const SelfReviewListingHeader = [
+    'Projects',
+    'From date',
+    'To date',
+    'Due Fom Date',
+    'Status'
+  ]
   const peerReviews = useSelector(
     state => state.peerReviewReducer.userPeerReview
   )
+  const userSelfReviews = useSelector(
+    state => state.selfReviewReducer.userSelfReviewDeatils
+  )
+  const projectData = useSelector(state => state.projectReducer.projects)
+  // const employeeData = useSelector(state => state.EmployeeInfo.employeeData)
 
   useEffect(() => {
-    dispatch(loadAllPeerForUser())
+    dispatch(loadAllPeerForUser(currentUser._id))
+    dispatch(loadAllSelfReviewsForUser(currentUser.employee_id))
   }, [dispatch])
 
+  useEffect(() => {
+    if (userSelfReviews && userSelfReviews.length > 0) {
+      // dispatch(loadAllEmployeeData())
+      dispatch(loadAllProjects())
+    }
+  }, [dispatch, userSelfReviews])
+
   const tempArr = []
   let filteredEmployee
   if (peerReviews) {
-    filteredEmployee = peerReviews.data.data.filter(
+    filteredEmployee = peerReviews.filter(
       cls =>
-        cls.employee_under_review
-          .toLowerCase()
-          .includes(searchText.toLowerCase().trim()) &&
+        cls.employee_under_review.firstname +
+          ' ' +
+          cls.employee_under_review.lastname
+            .toLowerCase()
+            .includes(searchText.toLowerCase().trim()) &&
         cls.status !== 'Done' &&
         cls.status !== 'Inactive'
     )
     filteredEmployee.map((review, key) => {
       tempArr.push([
-        review.employee_under_review,
-        review.project,
+        review.employee_under_review.firstname +
+          ' ' +
+          review.employee_under_review.lastname,
+        review.project.title,
         review.to_date.slice(0, 10),
         review.status
       ])
       return 1
     })
   }
+  const userReviewDetailsArr = []
+  let projects
+  let projectsArr = []
+  if (
+    userSelfReviews &&
+    userSelfReviews.length > 0 &&
+    userReviewDetailsArr.length == 0 &&
+    projectData
+  ) {
+    userSelfReviews.map((review, key1) => {
+      projects = review.project_ids.split(',')
+      projectData.map((item, key) => {
+        if (projects[key] && item._id == projects[key].trim()) {
+          projectsArr.push([item.title])
+        }
+      })
+      userReviewDetailsArr.push([
+        projectsArr.join('\n'),
+        review.from_date.slice(0, 10),
```

Comment:
can you format date at server side? or use some helper

Reply:
Added helper function